src/components/plugins/SchemeRegistrar/Plugin.tsx

```diff
@@ -0,0 +1,118 @@
+import * as React from "react";
+import { SchemeRegistrarPlugin as Entity } from "@dorgtech/arc.js";
+import { CreateContextFeed } from "../../../runtime/ContextFeed";
+import {
+  Arc as Protocol,
+  ArcConfig as ProtocolConfig,
+  Component,
+  ComponentLogs,
+  ComponentProps,
+  PluginEntity,
+  PluginData,
+  Plugin,
+} from "../../../";
+
+interface RequiredProps extends ComponentProps {
+  // Plugin ID
+  id?: string | Entity;
+}
+
+interface InferredProps extends RequiredProps {
+  config: ProtocolConfig;
+  id: string | Entity;
+}
+
+class InferredPluginManager extends Component<
+  InferredProps,
+  PluginEntity,
+  PluginData
+> {
+  protected createEntity(): PluginEntity {
+    const { config, id } = this.props;
+
+    if (!config) {
+      throw Error(
+        "Arc Config Missing: Please provide this field as a prop, or use the inference component."
+      );
+    }
+
+    const pluginId = typeof id === "string" ? id : id.id;
+    return new Entity(config.connection, pluginId);
+  }
+
+  public static get Entity() {
+    return CreateContextFeed(
+      this._EntityContext.Consumer,
+      this._LogsContext.Consumer,
+      "PluginManager"
```

Comment:
Let's rename to SchemeRegistrar and use the name PluginManager for the SchemeFactory

Reply:
Shouldn't we remove the word `Scheme` from the names? @dOrgJelli